PyRDF/backend/Spark.py

```diff
@@ -0,0 +1,67 @@
+from __future__ import print_function
+from .Dist import Dist
+from pyspark import SparkConf, SparkContext
+
+class Spark(Dist):
+    """
+    Backend that executes the computational graph
+    using using `Spark` framework for distributed
+    execution.
+
+    """
+    def __init__(self, config = {}):
+        """
+        Creates an instance of the Spark
+        backend class.
+
+        Parameters
+        ----------
+        config : dict (optional)
+            The config options for Spark backend. The default
+            value is an empty Python dictionary `{}`. Config
+            specifically reads three options, 'SparkContext'
+            or 'SparkConf' and 'npartitions'. If 'SparkContext'
+            is not given, 'SparkContext' uses 'SparkConf' option,
+            which by default is 'None' and 'npartitions' is
+            '2' by default.
+
+        """
+        super(Spark, self).__init__(config)
+
+        try:
+            # Remove the value of 'npartitions' from config dict
+            self.npartitions = config.pop('npartitions')
+        except KeyError:
+            # Set the default value to 2 if 'npartitions'
+            # doesn't exist
+            self.npartitions = 2
+
```

Comment:
@etejedor and mentors, I feel its not very clean to pop out only `npartitions` from `config` and set the rest as `SparkConf`. What do you guys think ?

Reply:
Why?
Other thing, what was discussed was that if no `npartitions` was provided, you would take the value of number of executors, and not a random 2.

Reply:
@diocas , me and @etejedor agreed that the value has to be `2` from `spark.executor.instances` in this link : [https://spark.apache.org/docs/latest/running-on-yarn.html#spark-properties](https://spark.apache.org/docs/latest/running-on-yarn.html#spark-properties) . And the `2` is compulsorily required here because, we have to create the range pairs only based on that

Reply:
If you agreed, it's ok.

But my point was (and what I took note from the meeting):
 * the user has `npartition`? use it.
 * Otherwise use the number of executors from SparkConf
    * This value does not exist, therefore is 2 (the default one)
    * This value exists, then it can be used

Here you pick a random 2, that might not be the real value of executors. Remember that ` SparkContext.getOrCreate` can pick an already existing `SparkContext` that contains a different `SparkConf`...

Reply:
@diocas , That's a good point. How about just taking the number of partitions as `spark.executor.instances` in the `config` dict ? That is how `SparkConf` stores it

Reply:
And this way, I don't have to pop it out of the `config` dict as well. I can directly pass it on to `SparkConf`

Reply:
@diocas , addressed your issue now. I misunderstood it earlier, but fixed it now. Thanks for pointing out :)

Reply:
I agree with @diocas that you should try to set the npartitions to the number of executors of the SparkContext. The logic you have now is good.